File: 0453-minimum-moves-to-equal-array-elements/0453-minimum-moves-to-equal-array-elements.go
func minMoves(nums []int) int {
    return sumOf(nums) - len(nums) * minOf(nums)
}

func sumOf(nums []int) int {
    sum := 0
    for _, n := range nums {
        sum += n
    }
    return sum
}

func minOf(nums []int) int {
    min := nums[0]
    for _, n := range nums[1:] {
        if n < min {
            min = n
        }
    }
    return min
}

File: 0453-minimum-moves-to-equal-array-elements/0453-minimum-moves-to-equal-array-elements.py
"""
class Solution:
    def minMoves(self, nums: List[int]) -> int:
        if len(nums) <= 1:
            return 0
        count = 0
        while not self.all_equal(nums):
            nums.sort()
            for i in range(len(nums)-1):
                nums[i] += 1
            count += 1    
        return count    
                
    def all_equal(self, nums: List[int]) -> bool:
        for i in range(1, len(nums)):
            if nums[i] != nums[i-1]:
                return False
        return True    
"""

class Solution:
    def minMoves(self, nums: List[int]) -> int:
        return sum(nums) - len(nums) * min(nums)
        